pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixServerStarter.java

```diff
@@ -321,6 +323,10 @@ public void start()
     LOGGER.info("Initializing server instance and registering state model factory");
     Utils.logVersions();
     ControllerLeaderLocator.create(_helixManager);
+
+    // Invoke Pinot Environment Provider's Init method to register the environment provider
+    PinotEnvironmentProviderFactory.init(_serverConf.subset(PREFIX_OF_CONFIG_OF_ENVIRONMENT_PROVIDER_FACTORY));
```

Comment:
This inits the provider. can we also do getEnvironment to override the configs ?

Reply:
Also what happens to existing setups with no provider. we should skip in case someone is not interested in setting up the provider.

Reply:
Done.